pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -0,0 +1,157 @@
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("tektoninstallation-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource TektonInstallation
+	if err := c.Watch(&source.Kind{Type: &v1alpha1.TektonInstallation{}}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}); err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource
+	enqueueRequestForOwner := &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &toolchainv1alpha1.TektonInstallation{},
+	}
+
+	return c.Watch(&source.Kind{Type: &olmv1alpha1.Subscription{}}, enqueueRequestForOwner)
+}
+
+// blank assignment to verify that ReconcileTektonInstallation implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileTektonInstallation{}
+
+// ReconcileTektonInstallation reconciles a TektonInstallation object
+type ReconcileTektonInstallation struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a TektonInstallation object and makes changes based on the state read
+// and what is in the TektonInstallation.Spec
+func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling TektonInstallation")
+
+	// Fetch the TektonInstallation instance
+	tektonInstallation := &toolchainv1alpha1.TektonInstallation{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: toolchain.TektonInstallation}, tektonInstallation); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		} else {
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, err
+		}
+	}
+
+	if err := r.EnsureTektonSubscription(reqLogger, tektonInstallation); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileTektonInstallation) EnsureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation) error {
+	tektonSubNamespace := tekton.SubscriptionNamespace
+	if err := r.ensureTektonSubscription(logger, tektonSubNamespace, tektonInstallation); err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, tektonInstallation, r.setStatusTektonSubscriptionFailed, err, "failed to create tekton subscription in namespace %s", tektonSubNamespace)
+	}
+	return r.StatusUpdate(logger, tektonInstallation, r.setStatusTektonSubscriptionReady, tekton.SubscriptionSuccess)
+}
+
+func (r *ReconcileTektonInstallation) ensureTektonSubscription(logger logr.Logger, ns string, tektonInstallation *v1alpha1.TektonInstallation) error {
```

Comment:
Minor: for consistency, we can move `ns` from place 2 to 3 in input args.

Reply:
Addressed in 31d139b